integration_tests/sdk/conftest.py

```diff
@@ -126,10 +127,14 @@ def engine(request, pytestconfig):
             pytest.skip(
                 "Skipped. Tests are only running against compute %s." % cmdline_compute_flag
             )
-
     # Test cases process the aqueduct engine as None. We do the conversion here
     # because fixture parameters are printed as part of test execution.
-    return request.param if request.param != "aqueduct_engine" else None
+    if request.param != "aqueduct_engine":
+        lazy_config = lazy_configuration(request.param)
+        global_config({"engine": request.param, "lazy": lazy_config})
+        return request.param
+    else:
+        return None
```

Comment:
I was actually thinking we could make a separate fixture that is autoused for this (so the user doesn't have to explicitly include `engine` for it to trigger). This fixture would essentially perform this logic here, yield, and then reset the global configuration at the end.

This makes the fixture more understandable I think, since it's focus is just to handle the global config lifecycle. It also makes it so that tests that don't include `engine` can benefit from this.

Reply:
Also, if we don't reset at the end, we may not be able to handle parallelized tests with multiple compute engines (eg. aqueduct_engine and k8s), since the former must have previews enabled and the second may not.

Reply:
By this, you mean just the global_config({"lazy"}) part? So the new fixture would just focus on lazy execution and I'd keep the logic for the engine here.

Reply:
mmm I was thinking everything, both the engine and config. Essentially, this fixture translates what we have in `test-config.yml` to the appropriate global_config() state. I think it's better to separate all that from the engine fixture, which has to be included explicitly by the user.